src/onchain/swap.rs

```diff
@@ -282,6 +284,27 @@ impl Swap {
             })
             .collect())
     }
+
+    /// Calculate the actual LSP fee for the given amount of a swap.
+    /// If the already existing inbound capacity is enough, no new channel is required.
+    ///
+    /// Parameters:
+    /// * `amount_sat` - amount in sats to compute LSP fee for
+    ///
+    /// Requires network: **yes**
+    pub fn calculate_swap_lsp_fee_for_amount(
+        &self,
+        amount_sat: u64,
+    ) -> Result<CalculateLspFeeResponse> {
+        calculate_swap_lsp_fee_for_amount(&self.support, amount_sat)
+    }
+}
+
+pub fn calculate_swap_lsp_fee_for_amount(
+    support: &Support,
+    amount_sat: u64,
+) -> Result<CalculateLspFeeResponse> {
+    support.calculate_lsp_fee_for_amount(amount_sat, Some(TWO_WEEKS))
```

Comment:
Do we need this function? It seems to me that everywhere it is needed we have an instance of `Swap`, so we can simply use the method

Reply:
Not really. But I introduced `Swap` where it was missing: 5e420e731955c7579898a75f26761c19658991a0